CIP-0043/README.md

```diff
@@ -0,0 +1,196 @@
+      big-endian unsigned integer.
+    * The last 32 bytes are the bytes of `s`.
+
+The builtin operations will error with a descriptive message if given inputs
+that don't correspond to the constraints above, return `False` if the signature
+fails to verify the input given the key, and `True` otherwise.
+
+## Rationale
+
+Implementing these functions as direct builtins in Plutus Core permits a
+workflow involving interoperability services like Wanchain and Renbridge. An
+example of such a workflow would be: given some Cardano assets, their Bitcoin
+equivalent could be minted, but only if the correct Schnorr signature can be
+provided. 
+
+We consider the implementation trustworthy: `secp256k1` is the reference
+implementation for both signature schemes, and is already being used in
+production by Bitcoin. Specifically, ECDSA signatures over the SECP256k1 curve
+were used by Bitcoin before Taproot, while Schnorr signatures over the same
+curve have been used since Taproot.
+
+An alternative approach could be to provide low-level primitives, which would
+allow any signature scheme (not just the ones under consideration here) to be
+implemented by whoever needs them. While this approach is certainly more
+flexible, it has two significant drawbacks:
+
+* It requires 'rolling your own crypto', rather than re-using existing
+  implementations. This has been shown historically to be a bad idea;
+  furthermore, if existing implementations have undergone review and audit, any
+  such re-implementations would give us the same assurances as those that have
+  been reviewed and audited.
+* It would be significantly costlier, as the computation would happen in Plutus
+  Core. Given the significant on-chain size restrictions, this would likely be
+  too costly for general use: many such schemes rely on large precomputed
+  tables, for example, which are totally unviable on-chain.
+
+It may be possible that some set of primitive can avoid both of these issues
+(for example, the suggestions in [this
+CIP](https://github.com/cardano-foundation/CIPs/pull/220)); in the meantime,
+providing direct support for commonly-used schemes such as these is worthwhile.
+
+## Backward Compatibility
+
+At the Plutus Core level, implementing this proposal induces no
+backwards-incompatibility: the proposed new primitives do not break any existing
+functionality or affect any other builtins. Likewise, at levels above Plutus
+Core (such as `PlutusTx`), no existing functionality should be affected.
+
+On-chain, this requires a hard fork.
+
+## Path to Active
+
+An implementation by MLabs already exists, and has been [merged into
+Plutus](https://github.com/input-output-hk/plutus/pull/4368). Tests of the
+functionality have also been included, although costing is currently
+outstanding, as it cannot be done by MLabs due to limitations in how costing is
+calculated. Costing will instead be done by the Plutus Core team.
```

Comment:
:+1: cc @michaelpj, I assume you are aware of this.

Reply:
yep